lib/driver.js

```diff
@@ -138,8 +142,10 @@ const MEMOIZED_FUNCTIONS = [
 ];
 
 class XCUITestDriver extends BaseDriver {
-  constructor (opts = {}, shouldValidateCaps = true) {
-    super(opts, shouldValidateCaps);
+  constructor (opts = {}, shouldValidateCaps = true, driverArgs = {}) {
+    opts = parseServerArgs(opts, driverArgs, Object.keys(ARGS_CONSTRAINTS), ARGS_CONSTRAINTS);
```

Comment:
I can observe some redundancy in the function arguments - why do we extract keys of ARGS_CONSTRAINTS and then pass ARGS_CONSTRAINTS themselves if we could only pass ARGS_CONSTRAINTS once and then fetch keys inside of the function body?

Reply:
also calling of `parseServerArgs` in inherited drivers is probably also unnecessary since it increases the amount of boilerplate code. Can we move this call to the base driver completely? ARGS_CONSTRAINTS then could be moved to a overridable method of BaseDriver class, which by default could just return an empty object

I understand that now we cannot inherit from the "new" BaseDriver, so we could simply leave a TODO there

Reply:
Yes, in the future the idea would be that all this logic is in the BaseDriver, but since we can't update the drivers to extend from the new BaseDriver yet, we should use the current solution but add a TODO